MovingBlocks/DestinationSol
# NUI: Add new specialised KeyActivatedButton and UIWarnButton widgets to mimic existing UI behaviour

# Description
This pull request adds the `KeyActivatedButton` and `UIWarnButton` widgets to mimic the UI behaviour of the pre-existing `SolUiControl` buttons.
- A `KeyActivatedButton` widget will activate when a key is pressed. If the key is held, then it will act the same as if a left-mouse button was holding it down.
- A `UIWarnButton` can flash an orange colour to notify the user that an event has occured, such as when new items are present in the inventory. It is also used in the tutorial to indicate which buttons to click on.

The pull request also adds a separate button disabled texture (previously missing) and moves the MigTestScreen into a seperate test screens package.
![image](https://user-images.githubusercontent.com/24301287/132127386-f4c54049-a398-4ea6-81f8-5a190edb582a.png)

# Testing
- Start a new game, or continue an existing one
- Open the console with the \` key or `F1`
- Use the command `showNUIScreen engine:solButtonsTestScreen`
- Press the keys shown on the buttons to press the corresponding buttons
- Hold the keys shown on the buttons to hold the corresponding buttons down
- The `I` button should be flashing orange

# Notes
- This pull request depends on MovingBlocks/TeraNUI#63 for testing.
- These changes were included in #588 but are being submitted separately so that work can start on porting other UI screens to NUI
